sql/expression/function/date.go

```diff
@@ -489,3 +496,65 @@ func (c CurrDate) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 func (c CurrDate) WithChildren(children ...sql.Expression) (sql.Expression, error) {
 	return NoArgFuncWithChildren(c, children)
 }
+
+// Determines the return type of a DateAdd/DateSub expression
+// Logic is based on https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-add
+func getType(input sql.Expression, interval *expression.Interval) sql.Type {
```

Comment:
Keep in mind that packages share a namespace for functions, and this is a large package. And golang doesn't have function overloading.

Suggest something more specific here, like `intervalExprType(...)`

Reply:
We often omit `get` in functions names in golang

Reply:
Regarding 1: Engine tests will not catch this bug. I can add the below query to `QueryTests` in [enginetest/queries/queries.go](https://github.com/dolthub/go-mysql-server/blob/main/enginetest/queries/queries.go#L4714-L4717), but that test will pass with or without this fix. The data being returned is correct, we're just assigning it the wrong type. This is another case that would benefit from https://github.com/dolthub/dolt/issues/3646

```
{
  Query:    "SELECT DATE_ADD('2018-05-02 12:13:14', INTERVAL 1 day)",
  Expected: []sql.Row{{time.Date(2018, time.May, 3, 12, 13, 14, 0, time.UTC)}},
}, 
```

Regarding 2: looking into this

Reply:
Updating.